modules/apps/transfer/keeper/keeper.go

```diff
@@ -190,6 +191,40 @@ func (k Keeper) IterateDenomTraces(ctx sdk.Context, cb func(denomTrace types.Den
 	}
 }
 
+// setDenomMetadata sets an IBC token's denomination metadata
+func (k Keeper) setDenomMetadata(ctx sdk.Context, denomTrace types.DenomTrace) {
+	metadata := banktypes.Metadata{
+		Description: getMetadataDescription(denomTrace),
+		DenomUnits: []*banktypes.DenomUnit{
+			{
+				Denom:    denomTrace.BaseDenom,
+				Exponent: 0,
+			},
+		},
+		// Setting base as IBC hash denom since bank keepers's SetDenomMetadata uses
+		// Base as storeKey and the IBC hash is what gives this token uniqueness
```

Comment:
nit: might be nice to differentiate between store key and key path. 

```suggestion
		// Base as key path and the IBC hash is what gives this token uniqueness
```